src/libmongoc/tests/test-mongoc-uri.c

```diff
@@ -313,6 +313,59 @@ _auth_mechanism_username_required (const char *mechanism)
    }
 }
 
+static void
+_auth_mechanism_password_prohibited (const char *mechanism, const char *user_prefix, const char *uri_suffix)
+{
+   BSON_ASSERT_PARAM (mechanism);
+   BSON_ASSERT_PARAM (user_prefix);
+
+   bson_error_t error;
+
+   // None.
+   {
+      mongoc_uri_t *const uri = mongoc_uri_new_with_error (
+         tmp_str ("mongodb://localhost/?" MONGOC_URI_AUTHMECHANISM "=%s%s", mechanism, uri_suffix), &error);
```

Comment:
```suggestion
      mongoc_uri_t *const uri = mongoc_uri_new_with_error (
         tmp_str ("mongodb://%s@localhost/?" MONGOC_URI_AUTHMECHANISM "=%s%s", user_prefix, mechanism, uri_suffix),
         &error);
```

Suggest including `user_prefix`. IIUC this helper only intends to test changes in the password and keep the rest of the URI fixed.

If accepted, also update or remove the `ASSERT_CMPSTR (mongoc_uri_get_username (uri), NULL);` below.